### PR TITLE
Add blocks for messages.

### PR DESCRIPTION
**Pull request by nliutyi-hs**
This PR adds `message blocks`, which are passed by Slack in the `message_action` payload.
This will allow us to work with user messages in a much more sufficient way, removing the need to parse the Slack markdown text itself, but giving the flexibility to work with blocks.

We've used this code version in our QA env and there were no exceptions regarding to it in our runtime.

### CODE REVIEW

File: slack-base/src/main/java/com/hubspot/slack/client/models/blocks/messages/RichBlockStyle.java
Comment (motnyk) [inline slack-base/src/main/java/com/hubspot/slack/client/models/blocks/messages/RichBlockStyle.java:5]: This looks like `ListType` to me. `RichBlockStyle` suggests that this can be applied to any rich block but that isn't the case.

File: slack-base/src/main/java/com/hubspot/slack/client/models/blocks/messages/RichMessageBlock.java
Comment (motnyk) [inline slack-base/src/main/java/com/hubspot/slack/client/models/blocks/messages/RichMessageBlock.java:7]: I think that the block with the nested blocks is not a unique feature of message blocks. I'd probably renamed this to `HasNestedBlocks` or `BlockWithNestedBlocks`? Probably the former option is more common.

File: slack-base/src/main/java/com/hubspot/slack/client/models/blocks/messages/RichTextListIF.java
Comment (motnyk) [inline slack-base/src/main/java/com/hubspot/slack/client/models/blocks/messages/RichTextListIF.java:23]: What does it represent? I'm curious about the difference between 1 and 2 for example.

File: slack-base/src/main/java/com/hubspot/slack/client/models/blocks/messages/TextStyleIF.java
Comment (motnyk) [inline slack-base/src/main/java/com/hubspot/slack/client/models/blocks/messages/TextStyleIF.java:20]: Let's call it `isStrikethrough`?

Comment (motnyk) [reply slack-base/src/main/java/com/hubspot/slack/client/models/blocks/messages/TextStyleIF.java:20]: I don't see a reason to use the short name here.

File: slack-base/src/test/java/com/hubspot/slack/client/models/blocks/messages/MessageActionSerializationTest.java
Comment (motnyk) [inline slack-base/src/test/java/com/hubspot/slack/client/models/blocks/messages/MessageActionSerializationTest.java:32]: We parse unknown blocks into a separate "unknown" type, right? If so, does this mean that counting doesn't really test anything and we should compare the whole payload instead?

File: slack-base/src/main/java/com/hubspot/slack/client/models/blocks/messages/RichMessageBlock.java
Comment (motnyk) [reply slack-base/src/main/java/com/hubspot/slack/client/models/blocks/messages/RichMessageBlock.java:7]: Also, QQ, I remember there was some failure because of this additional type when we were not able to correctly parse a new type returned by Slack if it isn't mapped to a model in this client.
Is the same thing going to happen now?

**Review by motnyk — CHANGES_REQUESTED** [review]
@nliutyi-hs 👍  
One small suggestion, I'd really appreciate it if you add an example payload to simplify the PR review.

File: slack-base/src/test/java/com/hubspot/slack/client/models/blocks/messages/MessageActionSerializationTest.java
Comment (nliutyi-hs) [reply slack-base/src/test/java/com/hubspot/slack/client/models/blocks/messages/MessageActionSerializationTest.java:32]: Makes sense.

**Comment by nliutyi-hs** [comment]
@motnyk Regarding the example payload, I thought that `message_action.json` is a good example, and there is no need to duplicate it, but I should've mentioned this in the PR's description.

File: slack-base/src/main/java/com/hubspot/slack/client/models/blocks/messages/RichBlockStyle.java
Comment (nliutyi-hs) [reply slack-base/src/main/java/com/hubspot/slack/client/models/blocks/messages/RichBlockStyle.java:5]: Makes sense.

File: slack-base/src/main/java/com/hubspot/slack/client/models/blocks/messages/RichMessageBlock.java
Comment (nliutyi-hs) [reply slack-base/src/main/java/com/hubspot/slack/client/models/blocks/messages/RichMessageBlock.java:7]: I think yeah, we can rename it to `BlockWithNestedBlocks`. 
The previous errors were related to  `MessageBlock`, which was more of a marker interface. 
I could not rely 100% on our test run, as we had only our services up, and it's not the complete data set to make a conclusion.
We definitely won't merge this PR today( gonna merge it the morning), so we will be able to catch an error if it happens again.

File: slack-base/src/main/java/com/hubspot/slack/client/models/blocks/messages/TextStyleIF.java
Comment (nliutyi-hs) [reply slack-base/src/main/java/com/hubspot/slack/client/models/blocks/messages/TextStyleIF.java:20]: Yes, we can do it.

**Comment by nliutyi-hs** [comment]
@motnyk Gonna simplify the payload and build the message blocks in the code, for decent comparison.

File: slack-base/src/main/java/com/hubspot/slack/client/models/blocks/messages/RichTextListIF.java
Comment (nliutyi-hs) [reply slack-base/src/main/java/com/hubspot/slack/client/models/blocks/messages/RichTextListIF.java:25]: Also, it's good you asked about this, I've added the optional offset field, which is present in a few `RichTextList` blocks, but not in all of them.

**Comment by motnyk** [comment]
> Regarding the example payload, I thought that message_action.json is a good example, and there is no need to duplicate it, but I should've mentioned this in the PR's description.

That's a good call, I haven't really gotten to it yet at a point when I realized that I missed an example to simplify the review.

File: slack-base/src/main/java/com/hubspot/slack/client/models/blocks/messages/RichMessageBlock.java
Comment (motnyk) [reply slack-base/src/main/java/com/hubspot/slack/client/models/blocks/messages/RichMessageBlock.java]: @nliutyi-hs Can you try and add a nested block with an random unknown type(e.g. blabla) and run a unit test or something? I guess that's something we can easily catch even before we deploy any code.

**Review by motnyk — COMMENTED** [review]
@nliutyi-hs Okay, so I think we'll be good to go here once you add a test that covers parsing of all the blocks and address https://github.com/HubSpot/slack-client/pull/304#discussion_r1028377841. Actually, I feel like that comment basically translates into "add another test for parsing unknown nested type".

**Comment by nliutyi-hs** [comment]
@motnyk PR is updated, please take a look once you'll be able to.

**Review by motnyk — APPROVED** [review]

**Pull request merged**
